packages/merkle-distributor/src/IpfsHelper.ts

```diff
@@ -0,0 +1,44 @@
+// This script adds some useful functions for dealing with IPFS.
+
+const ipfsApi = require("ipfs-http-client");
+const nodeFetch = require("node-fetch");
+
+const ipfs = ipfsApi({ host: "ipfs.infura.io", port: "5001", protocol: "https" });
+
+async function uploadFile(file: any) {
+  const { path } = await ipfs.add({ content: Buffer.from(JSON.stringify(file)) });
+  return path;
+}
+
+async function viewFile(fileHash: string) {
+  try {
+    const response = await nodeFetch(`"https://ipfs.infura.io:5001/api/v0/get?arg=${fileHash}`);
```

Comment:
I'm really just using the ipfs library for upload. for viewing, the infura portal is by far easier. I spent a few hours trying to get the `ipfs` object to return json objects using the `view` & `cat` methods but they were inconstant. Going directly to the infra portal worked the most easily.

I properly just need to spend more time trying to get this to work correctly. The other option would be to refractor away the IPFS library complete and change the `uploadFile` to _also_ use the infura ipfs portal. However, when I tried this it was more limited in _how_ the uploads could be done. In particular, it expected a file upload (from local directory) rather than a string/buffer, as the current implementation support. this somewhat janky hybrid was the result of pain in working with these libraries between uploading and viewing files.

Reply:
ok got it! yea that sucks, but probably not worth spending a lot more time on